cmc-ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/PartyMapper.java

```diff
@@ -0,0 +1,116 @@
+package uk.gov.hmcts.cmc.ccd.mapper;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.ccd.domain.CCDCompany;
+import uk.gov.hmcts.cmc.ccd.domain.CCDIndividual;
+import uk.gov.hmcts.cmc.ccd.domain.CCDOrganisation;
+import uk.gov.hmcts.cmc.ccd.domain.CCDParty;
+import uk.gov.hmcts.cmc.ccd.domain.CCDPartyType;
+import uk.gov.hmcts.cmc.ccd.domain.CCDSoleTrader;
+import uk.gov.hmcts.cmc.ccd.exception.MappingException;
+import uk.gov.hmcts.cmc.domain.models.party.Company;
+import uk.gov.hmcts.cmc.domain.models.party.Individual;
+import uk.gov.hmcts.cmc.domain.models.party.Organisation;
+import uk.gov.hmcts.cmc.domain.models.party.Party;
+import uk.gov.hmcts.cmc.domain.models.party.SoleTrader;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+@Component
+public class PartyMapper implements Mapper<CCDParty, Party> {
+
+    private final IndividualMapper individualMapper;
+    private final CompanyMapper companyMapper;
+    private final OrganisationMapper organisationMapper;
+    private final SoleTraderMapper soleTraderMapper;
+    private final AddressMapper addressMapper;
+    private final RepresentativeMapper representativeMapper;
+
+    @Autowired
+    public PartyMapper(IndividualMapper individualMapper, CompanyMapper companyMapper,
```

Comment:
add new lines for parameters please,
perhaps use a factory for creating these rather than injecting all of them in here?